src/api-handler.ts

```diff
@@ -127,34 +121,37 @@ export class ApiHandler implements MagicalClass {
      *
      * @param {*} args      Arguments
      * @returns {Promise}   Resolvable API provider promise
-     * @memberof ApiHandler
      */
     public async handleRequest(...args: any): Promise<IRequestResponse> {
         const prop = args[0];
-        const api = this.apiEndpoints[prop];
+        const endpointSettings = this.apiEndpoints[prop];
 
         const queryParams = args[1] || {};
         const uriParams = args[2] || {};
         const requestConfig = args[3] || {};
 
-        const uri = api.url.replace(/:[a-z]+/ig, (str: string) => (uriParams[str.substr(1)] ? uriParams[str.substr(1)] : str));
-
-        const requestData = {
-            ...queryParams,
-        };
+        const uri = endpointSettings.url.replace(/:[a-z]+/ig, (str: string) => (uriParams[str.substr(1)] ? uriParams[str.substr(1)] : str));
 
         let responseData = null;
 
-        responseData = await this.httpRequestHandler[api.method](uri, requestData, requestConfig);
+        const additionalRequestSettings = { ...endpointSettings };
```

Comment:
If it's an object, it will be just spread into an object, why is that? Can just do it the following

````
const additionalRequestSettings = endpointSettings 
```

Yet, maybe not required. Doing so, u will still unbind from actual *ref address, so can just assign it.

Reply:
It's pretty ugly one yes, but we can't remove url/methods by reference. What we need to do is to pass all props down except url & method since they'd overwrite Axios' ones

Reply:
I see, yeah then we could